security/src/main/java/org/kaazing/gateway/security/auth/BasicLoginModule.java

```diff
@@ -61,16 +61,25 @@ public void initialize(Subject subject, CallbackHandler callbackHandler,
     protected boolean doLogin() throws LoginException {
 
         if (!authenticationSchemeIsBasic()) {
-            return true;
+            throw new LoginException("Only able to handle Basic authentication scheme");
```

Comment:
It's unclear to me on whether we want this behavior or not.  

I believe, that this would imply that for multiple realms this would require the same challenge scheme?  What would be the implications of ignoring (return false) if the scheme is not set?  Maybe further login modules fail